services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -376,12 +377,13 @@ async def list_files(
 
     @s3_exception_handler(_logger)
     async def file_exists(self, bucket: S3BucketName, *, s3_object: str) -> bool:
-        """Checks if an S3 object exists"""
-        # SEE https://www.peterbe.com/plog/fastest-way-to-find-out-if-a-file-exists-in-s3
-        s3_objects, _ = await _list_objects_v2_paginated(
-            self.client, bucket, s3_object, max_total_items=EXPAND_DIR_MAX_ITEM_COUNT
-        )
-        return len(s3_objects) > 0
+        try:
+            await self.client.head_object(Bucket=bucket, Key=s3_object)
```

Comment:
so what about this:
https://www.peterbe.com/plog/fastest-way-to-find-out-if-a-file-exists-in-s3

Did you check whether this causes issues?